chains/heads/listener.go

```diff
@@ -135,10 +135,10 @@ func (l *listener[HTH, S, ID, BLOCK_HASH]) Connected() bool {
 func (l *listener[HTH, S, ID, BLOCK_HASH]) HealthReport() map[string]error {
 	var err error
 	if !l.ReceivingHeads() {
-		err = errors.New("Listener is not receiving heads")
+		err = errors.Join(err, errors.New("Listener is not receiving heads"))
```

Comment:
The var err is just declared above, and not used.
So it is really unnecessary to use Join here.
Anyways both New or Join work here, so can leave it whatever